solr/core/src/java/org/apache/solr/core/QuerySenderListener.java

```diff
@@ -102,4 +103,48 @@ public void close() {}
     }
     log.info("QuerySenderListener done.");
   }
+
+  private ArrayList<NamedList<Object>> queries2list(ArrayList<Object> queries) {
```

Comment:
I would bet logic like this exists for either SimpleOrderedMap or NamedList parsing elsewhere. SOLR-11914 has references to related methods however those methods are going to NamedList, not FROM NamedList. Somewhere in request parsing this has to happen.

Reply:
Is there anything specific you'd like us to change here? I'm not aware of any existing code we could invoke instead of having the new logic here.